### PR TITLE
0924 update

### PR DESCRIPTION
**Pull request by Yinita**
**Mainly Description**
1. Update VLLM method to replace transformers.
2. fixed template.
3. added api cost calculate.
4. updated azure api call.

**Checklist:**
- [✔] I have added description accordingly.
- [✔ ] This PR is being made to `dev branch` AND NOT TO `main branch`.

### CODE REVIEW

**Comment by Yinita** [comment]
> @Yinita please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree company="Microsoft"

**Review by Yinita — COMMENTED** [review]
Update some methods and add chat template.

File: RecLM-eval/call_models/huggingface_models.py [hunk 1]
```diff
@@ -109,14 +104,31 @@ def run_chat(local_gpu_rank, model_path_or_name, question_file, answer_file, arg
...
+        system_prompt
+    )
+
     if args.rank == 0:
         pbar = tqdm(total=len(test_dataset), desc="inferencing")
         result_lists = []
     
     with torch.no_grad():
         for raw_result in generator(test_dataset, generation_config=greedy_config, batch_size=args.batch_size):
+            # 0920 note: here we need to edit to chat_template
```
Comment (Leavingseason) [inline RecLM-eval/call_models/huggingface_models.py:129]: Something TODO left here?

File: RecLM-eval/call_models/vllm_models.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+import os
+from tqdm import tqdm
+import json
+from transformers import AutoConfig, AutoTokenizer
+import torch
+from torch.utils.data import Dataset, DataLoader
+import vllm
+import functools
+
+def map_gpu_count(m):
```
Comment (Leavingseason) [inline RecLM-eval/call_models/vllm_models.py:13]: 如果有复用别人的代码，记得在注释里面注明出处。

File: RecLM-eval/eval.py [hunk 3]
```diff
@@ -1,19 +1,24 @@
-# Copyright (c) Microsoft Corporation.
+# Copyright (c) Microsoft Corporation./home/v-yitiahuang/.triton
```
Comment (Leavingseason) [inline RecLM-eval/eval.py:1]: what is this

File: RecLM-eval/evaluates/test.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+from azure.identity import get_bearer_token_provider, AzureCliCredential  
+from openai import AzureOpenAI 
+cost = 0
+credential = AzureCliCredential() 
+token_provider = get_bearer_token_provider( 
+        credential, 
+        "https://cognitiveservices.azure.com/.default") 
+client = AzureOpenAI( 
+        azure_endpoint="https://humanoidai.openai.azure.com/",
```
Comment (Leavingseason) [inline RecLM-eval/evaluates/test.py:11]: mask the urls

File: RecLM-eval/call_models/vllm_models.py (hunk 2 above)
Comment (Yinita) [reply RecLM-eval/call_models/vllm_models.py:13]: 这里都是我之前写的
